lib/reduce_quda.cu

```diff
@@ -128,6 +128,36 @@ namespace quda {
   };
 
   /**
+     Return the L1 norm of x
+  */
+  __device__ double norm1_(const double2 &a) { return fabs(a.x) + fabs(a.y); }
+  __device__ float norm1_(const float2 &a) { return (double)fabs(a.x) + (double)fabs(a.y); }
```

Comment:
I noticed this issue in the feature/multigrid branch, and it's fixed there.  Will apply the fix here too.